File: georocket-server/src/main/java/io/georocket/tasks/ImportingTask.java
package io.georocket.tasks;

/**
 * A task started by the {@link io.georocket.ImporterVerticle}
 * @author Michel Kraemer
 */
public class ImportingTask extends AbstractTask {
  private long importedChunks;

  /**
   * Package-visible default constructor
   */
  ImportingTask() {
    // nothing to do here
  }

  /**
   * Default constructor
   * @param correlationId the correlation ID this task belongs to
   */
  public ImportingTask(String correlationId) {
    super(correlationId);
  }

  /**
   * Get the number of chunks already imported by this task
   * @return the number of imported chunks
   */
  public long getImportedChunks() {
    return importedChunks;
  }

  /**
   * Set the number of chunks already imported by this task
   * @param importedChunks the number of imported chunks
   */
  public void setImportedChunks(long importedChunks) {
    this.importedChunks = importedChunks;
  }

  @Override
  public void inc(Task other) {
    if (!(other instanceof ImportingTask)) {
      throw new IllegalArgumentException("Illegal task type");
    }
    ImportingTask io = (ImportingTask)other;
    super.inc(other);
    setImportedChunks(getImportedChunks() + io.getImportedChunks());
  }
}

File: georocket-server/src/main/kotlin/io/georocket/tasks/ImportingTask.kt
package io.georocket.tasks

import io.georocket.util.UniqueID
import java.time.Instant

/**
 * A task started by the [io.georocket.ImporterVerticle]
 * @author Michel Kraemer
 */
data class ImportingTask(
  override val id: String = UniqueID.next(),
  override val correlationId: String,
  override val startTime: Instant = Instant.now(),
  override val endTime: Instant? = null,
  override val error: TaskError? = null,

  /**
   * The number of chunks already imported by this task
   */
  val importedChunks: Long = 0,

  /**
   * The total size of the file to be imported in bytes
   */
  val bytesTotal: Long,

  /**
   * The number of bytes completely processed from the file to be imported
   */
  val bytesProcessed: Long = 0
) : Task
